File: buff/classes/354/pub/ch8/Box.java
// javac Box.java && java Box

public class Box<Item> {

    private Item item;

    public void put(Item item) {
	this.item=item;
    }

    public Item get() {
	return item;
    }

    public static void main(String[] args) {
	Box<String> box=new Box<String>();
	String s="stuff";
	box.put(s);
	System.out.println(box.get());
    }
	
}

File: buff/classes/354/pub/ch8/Box.cc
// g++ -Wall -o Box Box.cc && ./Box

#include <iostream>
#include <string>

using namespace std;

#include "Box.h"

template <class Item> void Box<Item>::put(Item item) {
  this->item=item;
}

template <class Item> Item Box<Item>::get() {
  return item;
}

int main() {
  // or: Box<string> box;
  Box<string>* box=new Box<string>();
  string s="stuff";
  box->put(s);
  cout << box->get() << endl;
  return 0;
}
